File: LC_Gen_Codes/2827_setting.py
class Solution:
    def numberOfBeautifulIntegers(self, low: int, high: int, k: int) -> int:
        def isBeautiful(num):
            count_even = 0
            count_odd = 0
            while num > 0:
                digit = num % 10
                if digit % 2 == 0:
                    count_even += 1
                else:
                    count_odd += 1
                num = num // 10
            return count_even == count_odd
        
        count = 0
        for num in range(low, high+1):
            if num % k == 0 and isBeautiful(num):
                count += 1
                
        return count

File: LC_Gen_Codes_Java/2827_setting.java
Here is the Java implementation for the given code:

java
class Solution {
    public int numberOfBeautifulIntegers(int low, int high, int k) {
        int count = 0;
        
        for (int num = low; num <= high; num++) {
            if (num % k == 0 && isBeautiful(num)) {
                count++;
            }
        }
        
        return count;
    }
    
    private boolean isBeautiful(int num) {
        int count_even = 0;
        int count_odd = 0;
        
        while (num > 0) {
            int digit = num % 10;
            
            if (digit % 2 == 0) {
                count_even++;
            } else {
                count_odd++;
            }
            
            num = num / 10;
        }
        
        return count_even == count_odd;
    }
}